tutorials/speaker_recognition/ASR_with_SpeakerDiarization.ipynb

```diff
@@ -0,0 +1,547 @@
+    "    plt.scatter(np.arange(len(signal)),signal,s=1,marker='o',c='k')\n",
+    "    if len(overlay_color):\n",
+    "        plt.scatter(np.arange(len(signal)),signal,s=1,marker='o',c=overlay_color)\n",
+    "    fig.suptitle(text, fontsize=16)\n",
+    "    plt.xlabel('time (secs)', fontsize=18)\n",
+    "    plt.ylabel('signal strength', fontsize=14);\n",
+    "    plt.axis([0,len(signal),-0.5,+0.5])\n",
+    "    time_axis,_ = plt.xticks();\n",
+    "    plt.xticks(time_axis[:-1],time_axis[:-1]/sample_rate);\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "plot the audio "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "show_figure(signal)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We start our demonstration by first transcribing the audio using our pretrained model `QuartzNet15x5Base-En` and use the CTC output probabilities to get timestamps for words spoken. We then later use these timestamps to get speaker label information using speaker diarizer model. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Download and load pretrained quartznet asr model"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#Load model\n",
+    "asr_model = nemo_asr.models.EncDecCTCModel.from_pretrained(model_name='QuartzNet15x5Base-En', strict=False)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Trascribe the audio "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "files = [AUDIO_FILENAME]\n",
+    "transcript = asr_model.transcribe(paths2audio_files=files)[0]\n",
+    "print(f'Transcript: \"{transcript}\"')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Get CTC log probabilities with output labels"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# softmax implementation in NumPy\n",
+    "def softmax(logits):\n",
+    "    e = np.exp(logits - np.max(logits))\n",
+    "    return e / e.sum(axis=-1).reshape([logits.shape[0], 1])\n",
+    "\n",
+    "# let's do inference once again but without decoder\n",
+    "logits = asr_model.transcribe(files, logprobs=True)[0].cpu().numpy()\n",
+    "probs = softmax(logits)\n",
+    "\n",
+    "# 20ms is duration of a timestep at output of the model\n",
+    "time_stride = 0.02\n",
+    "\n",
+    "# get model's alphabet\n",
+    "labels = list(asr_model.decoder.vocabulary) + ['blank']\n",
+    "labels[0] = 'space'"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Get timestamps for blank symbols, spaces and words "
```

Comment:
A small explanation here would be helpful:
"We use CTC labels for voice activity detection. To detect speech and non-speech segments in the audio, we use blank and space labels in the CTC outputs. Consecutive labels with spaces or blanks longer than a threshold are considered non-speech segments" .